prow/gen-private-jobs.sh

```diff
@@ -41,20 +41,31 @@ go run ./genjobs \
   --labels preset-enable-ssh=true \
   --job-type postsubmit \
   --repo-whitelist istio \
-  --job-whitelist release_istio_postsubmit
+  --job-whitelist release_istio_postsubmit,release_istio_release-1.4_postsubmit
 
 # istio/istio test jobs(s) - presubmit(s) and postsubmit(s)
 go run ./genjobs \
   "${COMMON_OPTS[@]}" \
   --job-type presubmit,postsubmit \
   --repo-whitelist istio \
-  --job-blacklist release_istio_postsubmit
+  --job-blacklist release_istio_postsubmit,release_istio_release-1.4_postsubmit
 
-# istio/proxy build jobs(s) - postsubmit(s)
+# istio/proxy master build jobs(s) - postsubmit(s)
 go run ./genjobs \
   "${COMMON_OPTS[@]}" \
+  --branches master \
+  --labels preset-enable-netrc=true \
   --job-type postsubmit \
-  --env GCS_BUILD_BUCKET=istio-private-build,GCS_ARTIFACTS_BUCKET=istio-private-artifacts,DOCKER_REPOSITORY=istio-prow-build/envoy \
+  --env GCS_BUILD_BUCKET=istio-private-build,GCS_ARTIFACTS_BUCKET=istio-private-artifacts,DOCKER_REPOSITORY=istio-prow-build/envoy,ENVOY_REPOSITORY=https://github.com/envoyproxy/envoy-wasm,ENVOY_SHA=bc93450483712189f22b2225f76039d5fe1f8ff9,ENVOY_PREFIX=envoy-wasm \
```

Comment:
in a case where we are updating envoy for  security patch (or even normal sync) the flow would be:

- update istio/envoy
- update sha here
- then trigger istio/proxy update with a dummy change?

Reply:
when we send a patch to envoy we will get some new SHA which will eventually be the same one as OSS, so we DO actually want to update the istio/proxy directly I would think rather than through here. I think what we want to override is just the source for where we pull that DHA from?

Reply:
> when we send a patch to envoy we will get some new SHA which will eventually be the same one as OSS, so we DO actually want to update the istio/proxy directly I would think rather than through here. I think what we want to override is just the source for where we pull that DHA from?

I am not sure exactly what the patch flow is for `envoy`. Since we are mounting a `.netrc` file with credentials to clone private repos, we can actually specify the archive, sha, etc. here via *environment vars* OR directly in the `WORKSPACE` of the fork. Both are possible. 

https://github.com/istio/test-infra/blob/b36a8c1d76f15ba41b9337ae4cdb77280cc86ea9/prow/gen-private-jobs.sh#L66

Reply:
I think we can leave the SHA part to the proxy repo, and just override the repo or whatever that contains that Sha. Would that make sense @piotrsikora?

Reply:
Well, we're cloning the repository here, so we need to provide some commit to checkout.

But hardcoding it here feels a bit weird, would we need to update definitions of jobs in public(?) for each update?

Reply:
We aren't cloning the Envoy repo anywhere here are we? I thought that was done by bazel? So the SHA to checkout would be determined by the WORKSPACE file like today, and we just pass in the override for the repo?

Reply:
No, we're cloning it outside of Bazel, and then pointing Bazel at local checkout.
See: istio/proxy#2555.

Also, the private security fixes committed to istio-private will have a new SHA by definition, so retaining SHA from the `WORKSPACE` wouldn't work anyway

Reply:
@howardjohn 
Here is my proposed change in `istio/proxy`: https://github.com/istio/proxy/pull/2555/files. We pass in `url` (i.e. repo) and `sha`.

Reply:
I see, I missed that part. What I was thinking is in oss:
1. make changes to envoy
2. update sha in proxy workspace

so in private it should also be
1. make changes to envoy
2. update sha in proxy workspace

that way when everything is done we can directly push all the changes up stream.

Can we have the script read the sha from the workspace?

Reply:
I was under the impression that patches to `envoy` would be applied to a private repo (e.g. `istio-private/envoy`). Is that not the case?

Reply:
That's a good point, you're right.

Reply:
I updated my comment above to be more clear hopefully. Everything should be done in the private org then once embargo lifts we push everything from private to public

Reply:
That's the case, but once embargo is lifted, those commits are going to be pushed to OSS (retaining history and commit SHAs).

Reply:
Updated mine as well. Like I mentioned before, both approaches are possible.

Reply:
One downside (or thing to remember) is the `url` (i.e. repo) will need to be changed back to *public* value before merged back to *public* (which is not particularly clean). Using env vars does allow us to point to any arbitrary source without source code modification. Not sure  the best approach off the top of my head.

Reply:
If every SHA referenced by istio-private/proxy exists in istio-private/envoy we do not need to change it right? I think that would be the case because we would sync the repos? Unless I am misunderstanding when we get into a situation when we point to something that doesn't exist

Reply:
The `sha` will be the same but the `url` will be different, right?

Instead of:
`https://github.com/istio/envoy/archive/`

It will be:
`https://github.com/istio-private/envoy/archive/`

in `WORRKSPACE`. So when we merge back to `public` need to make sure `url` is the public one (is what I was getting at).

https://github.com/istio/proxy/blob/0758acc9ec49ae48193a43ff6d86572968ae00e7/WORKSPACE#L51

Reply:
t should not be difficult to remedy. Let's sync up tomorrow and briefly discuss to ensure we are on the same page.

Reply:
My suggestion was that we keep everything in this PR except the SHA part (`ENVOY_SHA=429ee52b1f45b57420db15f434a36ad9efd81828`) but we still pass the stuff like `ENVOY_REPOSITORY=https://github.com/istio-private/envoy`

Then we keep everything from https://github.com/istio/proxy/pull/2555 except instead of reading `ENVOY_SHA` as an environment variable we grab it from https://github.com/istio/proxy/blob/0758acc9ec49ae48193a43ff6d86572968ae00e7/WORKSPACE#L41

That way everything done in the istio-private/proxy repo will be identical to what we will eventually push to public

Reply:
`WORKSPACE` will always point to the public repo, i.e. `https://github.com/istio/envoy/archive/` and we're going to use `ENVOY_REPOSITORY` to override that in private builds, but the SHA is going to be the same in both repos, so we don't need to override it.

Reply:
Okay, I think we are on the same page. As it stands now with https://github.com/istio/proxy/pull/2555 all of `ENVOY_REPOSITORY, ENVOY_SHA, ENVOY_PREFIX` need to be set for it to work properly. 

What you are suggesting is that we only override `ENVOY_REPOSITORY` *here* and add custom logic to script [proxy-common.inc](https://github.com/istio/proxy/blob/master/prow/proxy-common.inc extract) to read remaining values from the `WORKSPACE`? That should be doable

Reply:
:+1: sounds great

Reply:
https://github.com/istio/proxy/pull/2556